RecoEgamma/ElectronIdentification/plugins/ElectronMVANtuplizer.cc

```diff
@@ -178,7 +168,15 @@ ElectronMVANtuplizer::ElectronMVANtuplizer(const edm::ParameterSet& iConfig)
   nValMaps_              (valMapBranchNames_.size()),
   mvaCatTags_            (iConfig.getUntrackedParameter<std::vector<std::string>>("eleMVACats")),
   mvaCatBranchNames_     (iConfig.getUntrackedParameter<std::vector<std::string>>("eleMVACatLabels")),
-  nCats_                 (mvaCatBranchNames_.size())
+  nCats_                 (mvaCatBranchNames_.size()),
+  src_                   (consumes<edm::View<reco::GsfElectron> >(iConfig.getParameter<edm::InputTag>("src")),
+                          consumes<edm::View<reco::GsfElectron> >(iConfig.getParameter<edm::InputTag>("srcMiniAOD"))),
+  vertices_              (consumes<std::vector<reco::Vertex> >(iConfig.getParameter<edm::InputTag>("vertices")),
+                          consumes<std::vector<reco::Vertex> >(iConfig.getParameter<edm::InputTag>("verticesMiniAOD"))),
```

Comment:
in addition to using mayConsumes, I expected that the constructor here and below will use ```src_``` as a master.

Reply:
Ok! I didn't use the master token pattern in some cases, because the detection was also made independently before. But you are right, it's probably better to use it where possible.